src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ILGeneratorImpl.cs

```diff
@@ -576,7 +576,7 @@ public override void Emit(OpCode opcode, Type cls)
             ArgumentNullException.ThrowIfNull(cls);
 
             EmitOpcode(opcode);
-            WriteOrReserveToken(_moduleBuilder.GetTypeHandle(cls), cls);
+            WriteOrReserveToken(_moduleBuilder.GetTypeHandleForIL(cls), cls);
```

Comment:
It would be nice to have more consistent naming between `GetTypeHandle*` and `GetMethodHandle`

Maybe rename `GetTypeHandleForIL` to `TryGetTypeHandle` to make it clear that it may not actually give you the type handle for unbaked types (similar to GetMethodHandle/GetFieldHandle).

 And assert in `GetTypeHandle` that it is not called on unbaked types.